### PR TITLE
Introduce module for constants

### PR DESCRIPTION
**Pull request by erikkemperman**
As discussed [here](https://github.com/gulpjs/vinyl-fs/issues/158).

### CODE REVIEW

**Comment by phated** [comment]
Needs rebase

File: test/dest-modes.js [hunk 1]
```diff
@@ -331,14 +332,17 @@ describe('.dest() with custom modes', function() {
...
-
     var inputBase = path.join(__dirname, './fixtures');
     var inputPath = path.join(__dirname, './fixtures/wow/suchempty');
     var expectedBase = path.join(__dirname, './out-fixtures/wow');
     var expectedPath = path.join(__dirname, './out-fixtures/wow/suchempty');
     // NOTE: Darwin does not set setgid
-    var expectedDirMode = (isDarwin ? parseInt('777', 8) : parseInt('2777', 8)) & ~process.umask();
-    var expectedFileMode = parseInt('677', 8) & ~process.umask();
+    var expectedDirMode = constants.DEFAULT_DIR_MODE;
+    if (!isDarwin) {
```
Comment (phated) [inline test/dest-modes.js]: The custom mkdirp is implemented, do we still need this check?  Maybe it needs to be improved?

Comment (erikkemperman) [inline test/dest-modes.js]: I meant to look into that, but haven't had time... But yes, if the mkdirp change does what I hope it does then this should not be needed.

Comment (phated) [inline test/dest-modes.js]: Alright, I'll merge this and take a look

Comment (erikkemperman) [inline test/dest-modes.js]: Cool, thanks for review and merge!

**Pull request merged**